driver.py

```diff
@@ -0,0 +1,191 @@
+    """
+    
+    check(features)
+    install(features)
+
+    release_cmd = ["cargo", "build", "--release", "--examples"]
+    cmd = ["cargo", "build", "--examples"]
+
+    cargo_features = filter_cargo_features(features)
+    if cargo_features:
+        release_cmd = release_cmd + ["--features"] + cargo_features
+        cmd = cmd + ["--features"] + cargo_features
+
+    subprocess.run(release_cmd)
+    subprocess.run(cmd)
+
+    for f in features: 
+        if f == 'aby':
+            if 'c_front' in features:
+                subprocess.run(["./scripts/build_mpc_c_test.zsh"])
+            subprocess.run(["./scripts/build_mpc_zokrates_test.zsh"])
+            subprocess.run(["./scripts/build_aby.zsh"])
+
+def test(features):
+    """
+    Run cargo tests and any test cases in the feature list
+
+    Parameters
+    ----------
+        features : set of str
+            set of features required
+    """
+
+    build(features)
+
+    subprocess.run(["cargo", "test"], check=True)
+    subprocess.run(["./scripts/zokrates_test.zsh"], check=True)
+    subprocess.run(["./scripts/test_zok_to_ilp.zsh"], check=True)
+    subprocess.run(["./scripts/test_zok_to_ilp_pf.zsh"], check=True)
+    subprocess.run(["./scripts/test_datalog.zsh"], check=True)
+    subprocess.run(["python3", "./scripts/aby_tests/zokrates_test_aby.py"], check=True)
+
+    for f in features: 
+        if f == 'aby':
+            if 'c_front' in features:
+                subprocess.run(["python3", "./scripts/aby_tests/c_test_aby.py"], check=True)
+
+def format():
+    print("formatting!")
+    subprocess.run(["cargo", "fmt", "--all"], check=True)
+
+def lint():
+    print("linting!")
+    subprocess.run(["cargo", "clippy"], check=True)
+
+def clean():
+    print("cleaning!")
+    subprocess.run(["./scripts/clean_aby.zsh"])
+    subprocess.run(["rm", "-rf", "scripts/aby_tests/__pycache__"])
+    subprocess.run(["rm", "-rf", "P", "V", "pi", "perf.data perf.data.old flamegraph.svg"])
+
+def set_features(features):
+    """
+    Filter invalid features and save features to a file.
+
+    Parameters
+    ----------
+        features : set of str
+            set of features required
+    """
+    # reset features
+    if 'none' in features:
+        features = set()
+
+    def verify_feature(f):
+        if f in valid_features:
+            return True
+        return False
+    features = set(sorted([f for f in features if verify_feature(f)]))
+    save_features(features)
+    print("Feature set:", sorted(list(features)))
+    return features 
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser()
+    parser.add_argument("-i", "--install", action="store_true", help="install all dependencies from the feature set")
+    parser.add_argument("-c", "--check", action="store_true", help="run `cargo check`")
+    parser.add_argument("-b", "--build", action="store_true", help="run `cargo build` and build all dependencies from the feature set")
```

Comment:
Question: would it make more sense to have dependency building be part of `build` or `install`?

Reply:
Currently I have it as part of `install`. The reason is because in our old ci pipeline, we build the dependencies first, then cache the dependencies, then build circ. Would moving dependency building to build affect the ci?

Reply:
I agree that we should build (non-Rust) dependencies during `install`. Is that what `driver.py` does right now? I'll leave a comment where I think the dependency build it happening.

Reply:
yep! the comment is in the right spot.